cmake/Modules/FindCOIN.cmake

```diff
@@ -83,23 +86,44 @@ macro(_coin_find_lib _libname _lib_file_names _lib_file_names_debug)
       HINTS ${COIN_ROOT_DIR}/lib
     )
 
-    if(EXISTS "${COIN_INCLUDE_DIR}/${_libnamelower}/coin")
-      ## Unfortunately coin uses a weird include model and includes headers from different subpackages
-      ## via include<File.hpp> instead of include<coin/File.hpp>
-      set(_libspecific_include "${COIN_INCLUDE_DIR}/${_libnamelower}" "${COIN_INCLUDE_DIR}/${_libnamelower}/coin")
-    else()
-      set(_libspecific_include)
+    find_path(
+      ${_libname}_INCLUDE_DIR
+      NAMES 
+        ${FOLDER}/coin/${HNAME} 
+    )
+
+    set (INCLUDE_DIRS)
+    if (EXISTS "${${_libname}_INCLUDE_DIR}/${FOLDER}")
+      # we (in OpenMS) include the coin headers with e.g., coin/ClpInterface.h
+      list(APPEND INCLUDE_DIRS "${${_libname}_INCLUDE_DIR}/${FOLDER}")
+      # while coin-or includes headers from other sublibraries with just e.g., ClpModel.h
+      list(APPEND INCLUDE_DIRS "${${_libname}_INCLUDE_DIR}/${FOLDER}/coin")
     endif()
 
+    if (EXISTS ${COIN_INCLUDE_DIR})
+      list(APPEND INCLUDE_DIRS ${COIN_INCLUDE_DIR})
+    endif()
+
+    # Before, we handled the vcpkg specific sublibrary structure as follows. This should be handled equally
+    # by the more generic searching above (which also works for brew). But needs to be tested. I don't use
+    # vcpkg currently
+    # _libnamelower = FOLDER
+
+    #if(EXISTS "${COIN_INCLUDE_DIR}/${_libnamelower}/coin")
+    #  set(_libspecific_include "${COIN_INCLUDE_DIR}/${_libnamelower}" "${COIN_INCLUDE_DIR}/${_libnamelower}/coin")
```

Comment:
why did (and still do) both possibilities need to be added?

Reply:
See the description above the lines.
We use `include "coin/CoinUtils.h"`
They use `include "CoinUtils.h"` for internal reference to other sublibs.